### PR TITLE
Terms of Service

### PR DESCRIPTION
**Pull request by carycheng**
Thanks Karthik!

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @carycheng77 has signed the CLA. Thanks for the pull request!

File: src/main/java/com/box/sdk/BoxTermsOfServiceUserStatus.java [hunk 1]
```diff
@@ -0,0 +1,288 @@
...
+                                                                     String termsOfServiceID, String userID) {
+        final int limit = 2;
+        QueryStringBuilder builder = new QueryStringBuilder();
+        builder.appendParam("tos_id", termsOfServiceID);
+        if (userID != null) {
+            builder.appendParam("user_id", userID);
+        }
+
+        return new BoxResourceIterable<BoxTermsOfServiceUserStatus.Info>(api,
+                ALL_TERMS_OF_SERVICE_USER_STATUSES_TEMPLATE.buildWithQuery(api.getBaseURL(), builder.toString()),
```
Comment (KarthikShanmugasundaram) [inline src/main/java/com/box/sdk/BoxTermsOfServiceUserStatus.java]: Please move the statement as a separate one.

File: src/main/java/com/box/sdk/BoxTermsOfService.java [hunk 2]
```diff
@@ -0,0 +1,298 @@
...
+                    if (this.enterprise == null) {
+                        this.enterprise = new BoxEnterprise(jsonObject);
+                    } else {
+                        this.enterprise.update(jsonObject);
+                    }
+                } else if (memberName.equals("type")) {
+                    this.type = value.asString();
+                } else if (memberName.equals("tos_type")) {
+                    this.tosType = value.asString();
+                } else if (memberName.equals("text")) {
```
Comment (KarthikShanmugasundaram) [inline src/main/java/com/box/sdk/BoxTermsOfService.java]: value is mandatory field? if there is any optional fields, the null checks are done?

File: src/main/java/com/box/sdk/BoxTermsOfServiceUserStatus.java [hunk 3]
```diff
@@ -0,0 +1,288 @@
...
+                    JsonObject tosJSON = value.asObject();
+                    String termsOfServiceID = tosJSON.get("id").asString();
+                    BoxTermsOfService termsOfService = new BoxTermsOfService(getAPI(), termsOfServiceID);
+                    this.termsOfService = termsOfService.new Info(tosJSON);
+                } else if (memberName.equals("user")) {
+                    JsonObject userJSON = value.asObject();
+                    String userID = userJSON.get("id").asString();
+                    BoxUser user = new BoxUser(getAPI(), userID);
+                    this.user = user.new Info(userJSON);
+                } else if (memberName.equals("is_accepted")) {
```
Comment (KarthikShanmugasundaram) [inline src/main/java/com/box/sdk/BoxTermsOfServiceUserStatus.java]: if there is any optional fields, the null checks are done?

File: src/test/java/com/box/sdk/BoxTermsOfServiceTest.java [hunk 4]
```diff
@@ -0,0 +1,205 @@
...
+        }
+    }
+
+    @Test
+    @Category(IntegrationTest.class)
+    public void updateTermsOfServiceInfoSucceeds() {
+        BoxAPIConnection api = new BoxAPIConnection(TestConfig.getAccessToken());
+        String newStatus = "enabled";
+        String newText = "New text";
+        BoxTermsOfService termsOfService = new BoxTermsOfService(api, "2778");
```
Comment (KarthikShanmugasundaram) [inline src/test/java/com/box/sdk/BoxTermsOfServiceTest.java]: Please add more negative test cases

File: src/test/java/com/box/sdk/BoxTermsOfServiceUserStatusTest.java [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+
+        for (BoxTermsOfServiceUserStatus.Info info: tosUserStatusInfo) {
+            assertThat(info, is(notNullValue()));
+        }
+    }
+
+    @Test
+    @Category(IntegrationTest.class)
+    public void getUserStatusInfoOnTermsOfServiceWithUserIDSucceeds() {
+        BoxAPIConnection api = new BoxAPIConnection(TestConfig.getAccessToken());
```
Comment (KarthikShanmugasundaram) [inline src/test/java/com/box/sdk/BoxTermsOfServiceUserStatusTest.java]: Please add more negative test cases

File: src/main/java/com/box/sdk/BoxTermsOfService.java [hunk 6]
```diff
@@ -0,0 +1,298 @@
...
+                    this.text = value.asString();
+                } else if (memberName.equals("created_at")) {
+                    this.createdAt = BoxDateFormat.parse(value.asString());
+                } else if (memberName.equals("modified_at")) {
+                    this.modifiedAt = BoxDateFormat.parse(value.asString());
+                }
+            } catch (ParseException e) {
+                assert false : "A ParseException indicates a bug in the SDK.";
```
[the remark this answers is not in the document]
Comment (carycheng77) [reply src/main/java/com/box/sdk/BoxTermsOfService.java]: Hey @KarthikShanmugasundaram wondering comment we should put here. Reason being I followed how it was built in BoxRetentionPolicy, BoxDevicePin, and other files and the ParseException e is followed by: "A ParseException indicates a bug in in the SDK". Any input is greatly appreciated!

[the remark this answers is not in the document]
Comment (KarthikShanmugasundaram) [reply src/main/java/com/box/sdk/BoxTermsOfService.java]: For example: Terms of Service Parsing failed. {the message from parse exception to understand more on the error)

**Pull request merged**